src/main/java/com/glodblock/github/client/gui/base/FCBaseMEGui.java

```diff
@@ -42,16 +71,84 @@ public FCBaseMEGui(final InventoryPlayer inventoryPlayer, Container container) {
             if (target instanceof IWirelessTerminal
                     && ((IWirelessTerminal) target).getItemStack().getItem() instanceof ItemWirelessUltraTerminal) {
                 this.drawSwitchGuiBtn = true;
+                this.hasMagnetCard = Util.Wireless.hasMagnetCard(((IWirelessTerminal) target).getItemStack());
+                this.channel = ((IWirelessTerminal) target).getChannel();
+            }
+        }
+    }
+
+    @Override
+    public List<String> handleItemTooltip(final ItemStack stack, final int mouseX, final int mouseY,
+            final List<String> currentToolTip) {
+        if (this instanceof GuiFluidMonitor) {
+            if (stack != null && !isShiftKeyDown()) {
+                final Slot s = this.getSlot(mouseX, mouseY);
+                final boolean isSlotME = s instanceof SlotME;
+                if (isSlotME || s instanceof SlotFake) {
+                    final int BigNumber = AEConfig.instance.useTerminalUseLargeFont() ? 999 : 9999;
+
+                    IAEItemStack myStack = null;
+
+                    try {
+                        myStack = Platform.getAEStackInSlot(s);
+                    } catch (final Throwable ignore) {}
+
+                    if (myStack != null) {
+                        if (myStack.getStackSize() > BigNumber || (myStack.getStackSize() > 1)) {
+                            final String formattedAmount = NumberFormat.getNumberInstance(Locale.US)
+                                    .format(myStack.getStackSize());
+                            final String format = I18n.format(TT_FLUID_TERMINAL_AMOUNT, formattedAmount);
+                            currentToolTip.add("\u00a77" + format);
+                        }
+                    }
+                }
             }
+            return currentToolTip;
+        } else {
+            return super.handleItemTooltip(stack, mouseX, mouseY, currentToolTip);
+        }
+
+    }
+
+    protected ItemMagnetCard.Mode getMagnetMode() {
+        if (this.hasMagnetCard) {
+            return ((FCBaseContainer) this.inventorySlots).mode;
         }
+        return null;
+    }
+
+    protected boolean isRestock() {
+        return ((FCBaseContainer) this.inventorySlots).restock;
+    }
+
+    protected boolean isSyncData() {
+        return ((FCBaseContainer) this.inventorySlots).sync;
     }
 
     @Override
     public void onGuiClosed() {
         super.onGuiClosed();
     }
 
+    protected void setSyncState() {
+        if (this.inventorySlots instanceof FCContainerMonitor && !isPortableCell()) {
+            this.buttonList.add(
+                    this.dataSyncEnableBtn = new GuiFCImgButton(
```

Comment:
Let's split instance creation and it usage to add into list. Take a look `appeng.client.gui.implementations.GuiInterfaceTerminal.initGui` and adapt to new approach
- instance creation
- adding into list
- set correct coordinates using button chain and constants